docs/ImageMirrorSetDesign.md

```diff
@@ -0,0 +1,167 @@
+# Summary
+
+Users need a way to set up mirror configuration. Users can set mirror configurations using ImageContentSourcePolicy currently via MCO currently, but if they want to use mirror when pulling image with tag instead of digest in the pull spec, they need to follow this [solution](https://access.redhat.com/solutions/4817401) to update the registries configuration via MCO. We can hopefully help customers by documenting CRDs with the `ImageDigestMirrorSet` and `ImageTagMirrorSet` tuneables and have the MCO use these tuneables when rendering the registries.conf files to Ignition/disk.
+
+# Motivation
+
+The MCO contains the logic to write the registries.conf configuration files to Ignition. When Ignition starts on a machine it writes this file to configure the mirrors.
+
+## Goals
+
+1. Setting cluster wide mirror registries via `ImageDigestMirrorSet`. The mirrors will be used only when the pull spec is specified with a digest.
+
+2. Setting cluster wide mirror registries via `ImageTagMirrorSet`. The mirrors will be used only when the pull spec is specified with a tag.
+
+3. Setting cluster wide mirror registries, but blocking the image pull from the registry that specified as source in the pull spec by setting the enum field `mirrorSourcePolicy`. If all the mirror registries are failed, the image pull will be redirected to the registry in the pull spec by default.
+
+4.  Before the `ImageContentSourcePolicy` CRD gets deprecated, reject to update one of `ImageDigestMirrorSet`/ `ImageTagMirrorSet` and `ImageContentSourcePolicy` when there is an attempt to update both objects on the cluster.
+
+## Non-Goals
+
+Using by-tag configurations for OpenShift release images is not recommended. Those should still be referenced by digest.
+
+# Proposal
+
+Extend the ContainerRuntimeConfigController to support ImageDigestMirrorSet and ImageTagMirrorSet CRDs. Adding or deleting `ImageDigestMirrorSet` or `ImageTagMirrorSet` objects will add or delete the corresponding registry entries of registries.conf, see **containers-registries.conf(5)** for more information.
```

Comment:
nit: maybe add the link to [containers-registries.conf(5)](https://github.com/containers/image/blob/main/docs/containers-registries.conf.5.md) as well?